app/components/market/submit/submit-market-application.component.ts

```diff
@@ -1,208 +1,171 @@
-    }
-
-    private _computeType() {
-        this._computeDependsOn();
-
-        const param = this._param;
-        if (param.metadata && param.metadata.advancedType) {
-            const type = param.metadata.advancedType;
-            if (!ObjectUtils.values(NcjParameterExtendedType as any).includes(type)) {
-                log.error(`Advanced typed '${type}' is unkown!`, NcjParameterExtendedType);
-            }
-            this.type = type as NcjParameterExtendedType;
-            return;
-        }
-        this.type = param.type as any;
-    }
-}
-const ConventionNames = {
-    jobName: "jobName",
-};
 @Component({
     selector: "bl-submit-market-application",
     templateUrl: "submit-market-application.html",
 })
-export class SubmitMarketApplicationComponent implements OnInit {
+export class SubmitMarketApplicationComponent {
     public static breadcrumb() {
         return { name: "Submit" };
     }
+    private title;
+    private form;
+    private mode_state;
+    private applicationId;
+    private actionId;
+    private jobTemplate;
+    private poolTemplate;
+    private pickedPool = new FormControl(null);
+    private error;
 
-    public NcjParameterExtendedType = NcjParameterExtendedType;
-
-    public jobNameParam: NcjParameterWrapper;
-
-    public applicationId: string;
-    public actionId: string;
-    public title = "";
-    public jobTemplate: NcjJobTemplate;
-    public poolTemplate;
-    public form: FormGroup;
-    public jobFormGroup: FormGroup;
-    public pickedPool = new FormControl(null);
-    public otherParameters: NcjParameterWrapper[];
-    public error: ServerError;
-
-    private _formValue: any;
-
-    // Constructor create data structure from reading json files
     constructor(
         public formBuilder: FormBuilder,
         private pythonRpcService: PythonRpcService,
         private route: ActivatedRoute,
         private router: Router,
-        private templateService: NcjTemplateService) { }
-
-    public ngOnInit() {
+        private templateService: NcjTemplateService) {
+             this.mode_state = "None";
+             this.form = new FormGroup({a: new FormControl()});
+    }
+    private ngOnInit() {
         this.route.params.subscribe((params) => {
             this.applicationId = params["applicationId"];
             this.actionId = params["actionId"];
-            this._updateTitle();
-            this._getTemplates();
-        });
-        this.route.queryParams.subscribe((params) => {
-            if (params.formParams) {
-                try {
-                    const value = JSON.parse(params.formParams);
-                    this._formValue = value;
-                    this.form.setValue(value);
-                } catch (e) {
-                    log.warn("Invalid form param. Not valid json.", params.formParams as any);
-                }
-            }
+            this.templateService.getTemplates(this.applicationId, this.actionId).subscribe((templates) => {
```

Comment:
minor: why is this 'templates' instead of 'template' (note plural vs. singular')?